From 713560ec861536ce364fe1424c2693b2293a175a Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Wed, 24 Jul 2024 11:32:52 +0200
Subject: [PATCH 1045/1695] plat-stm32mp2: access BSEC only if CFG_STM32_BSEC3
 is set

BSEC is accessible either by the TDCID, or by the BootROM after a reset.
When the CFG_STM32_BSEC3 switch isn't embedded (i.e. when M33 is TDCID),
do not call BSEC functions.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 7526d4062d4d6482330d50794c175a681ae8a939 plat-stm32mp2: access BSEC only if CFG_STM32_BSEC3 is set]
Change-Id: I29de8733d54843e186d3186d95efde0453642ace
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/395778
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 core/arch/arm/plat-stm32mp2/main.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/core/arch/arm/plat-stm32mp2/main.c b/core/arch/arm/plat-stm32mp2/main.c
index a222ab234..770b2f1bb 100644
--- a/core/arch/arm/plat-stm32mp2/main.c
+++ b/core/arch/arm/plat-stm32mp2/main.c
@@ -208,8 +208,9 @@ early_init_late(init_debug);
 #endif /* CFG_STM32_BSEC3 */
 
 #ifdef CFG_STM32_CPU_OPP
-bool stm32mp_supports_cpu_opp(uint32_t opp_id)
+bool stm32mp_supports_cpu_opp(uint32_t opp_id __maybe_unused)
 {
+#ifdef CFG_STM32_BSEC3
 	static uint32_t part_number;
 	uint32_t otp = 0;
 	size_t bit_len = 0;
@@ -229,6 +230,9 @@ bool stm32mp_supports_cpu_opp(uint32_t opp_id)
 		id = BIT(0);
 
 	return (opp_id & id) == id;
+#else
+	return false;
+#endif
 }
 #endif /* CFG_STM32_CPU_OPP */
 
-- 
2.47.2

